mkdd_extender.py

```diff
@@ -2817,6 +2818,17 @@ def write_description_file(args: argparse.Namespace, added_course_names: 'list[s
             'functionality should follow the same structure; the game will apply the 10000 offset '
             'to the models\' geometry after the tilt rotation is applied.',
         ),
+        ('Extended Terrain Types', bool,
+         'If enabled, new terrain types that add custom mechanics will be added to the game. '
+         '\n\n'
+         'Information about these terrain types, including instructions on how to add them '
+         'to a Custom Track can be found at the following locations:'
+         '\n\n'
+         'https://lance-o.github.io/extended_terrain_types '
+         '\n\n'
+         'https://github.com/lance-o/extended_terrain_types '
+         '\n\n'
+         'NOTE: This patch does not support the NTSC-U Debug version.'),
```

Comment:
> Instead of adding a note for the unsupported NTSC-U Debug version, I think a runtime check can be added in code_patcher.py to throw an error if the code patch is enabled and NTSC-U Debug is used. It'd be preferable over silently failing to apply the code patch.

I've added a temporary function into `code_patcher.py` to throw a runtime error in the event the user tries to patch the Debug .iso with `Extended Terrain Types` checked.

This should _really_ be done from from `mkdd_extender.py` before any Custom Courses are compiled, but since you suggested `code_patcher.py` I put it there for now. Is there any reason it should be called from `code_patcher.py` that I'm not considering?

Reply:
> [...] Is there any reason it should be called from `code_patcher.py` that I'm not considering?

You are not wrong... Having the check early would stop the Extender from unpacking/processing a bunch of custom courses only to discard it all later.

The reason why I suggest adding it in `code_patches.py` is that that's the point where the missing functionality would need to be implemented in the future, and the file that would be modified in the future. I don't think many use the debug build anyway, so the fact that the check is not in the more performant point doesn't concern me much.

(Having said that, I don't mind if you move it to other place.)

Reply:
I was really intending on moving it, but if you don't mind then it's fine where it is. The amount of time it was taking me to compile was annoying me while I was testing the runtime error, since I had a lot of slightly different copies of test tracks, so I thought to suggest it.

Reply:
As I'm on Linux, where the file system is a lot faster, I never think of that as a problem. 

I'm happy for you to move it (especially if you don't think support for the debug build will ever be added).

I'll resolve the thread and leave it up to you.